solanaj/src/main/java/org/p2p/solanaj/kits/renBridge/BurnAndRelease.kt

```diff
@@ -0,0 +1,92 @@
+package org.p2p.solanaj.kits.renBridge
+
+import org.bitcoinj.core.Base58
+import org.p2p.solanaj.core.Account
+import org.p2p.solanaj.core.PublicKey
+import org.p2p.solanaj.kits.renBridge.renVM.RenVMProvider
+import org.p2p.solanaj.rpc.RpcException
+import org.p2p.solanaj.rpc.RpcSolanaInteractor
+import org.p2p.solanaj.utils.Hash
+import org.p2p.solanaj.utils.Utils
+import org.p2p.solanaj.utils.crypto.Hex
+import java.math.BigInteger
+import java.nio.ByteBuffer
+
+class BurnAndRelease(
+    private val renVMProvider: RenVMProvider,
+    private val rpcSolanaInteractor: RpcSolanaInteractor,
+    private val state: LockAndMint.State
+) {
+
+    private var nonceBuffer: ByteArray = byteArrayOf()
+    private var recepient: String = ""
+
+    suspend fun submitBurnTransaction(
+        account: PublicKey,
+        amount: String,
+        recepient: String,
+        signer: Account
+    ): BurnDetails {
+        this.recepient = recepient
+        return rpcSolanaInteractor.submitBurn(account, amount, recepient, signer)
+    }
+
+    fun getBurnState(burnDetails: BurnDetails, amount: String): LockAndMint.State {
+        val txId = Base58.decode(burnDetails.confirmedSignature)
+        nonceBuffer = getNonceBuffer(burnDetails.nonce)
+        val nHash = Hash.generateNHash(nonceBuffer, txId, "0")
+        val pHash = Hash.generatePHash()
+        val sHash = Hash.generateSHash("BTC/toBitcoin")
+        val gHash =
+            Hash.generateGHash(Hex.encode(Utils.addressToBytes(burnDetails.recepient)), Hex.encode(sHash), nonceBuffer)
+
+        val txHash = renVMProvider.getTxHash(
+            selector = RenVMProvider.BURN_SELECTOR,
+            gHash = gHash,
+            gPubKey = byteArrayOf(),
+            nHash = nHash,
+            nonce = nonceBuffer,
```

Comment:
nonceBuffer.copyOf()

Reply:
you mean nonceBuffer.copyOf(burnDetails.nonce) ?

Reply:
no, just copyOf()
it's a good practice to always return copies of byteArray instead of original